base_external_dbsource_mysql/models/base_external_dbsource.py

```diff
@@ -0,0 +1,37 @@
+# Copyright 2011 Daniel Reis
+# Copyright 2016 LasLabs Inc.
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl.html).
+
+import sqlalchemy
+
+from odoo import models
+
+from odoo.addons.base_external_dbsource.models import base_external_dbsource
+
+base_external_dbsource.BaseExternalDbsource.CONNECTORS.append(("mysql", "MySQL"))
+
+
+class BaseExternalDbsource(models.Model):
+    """It provides logic for connection to a MySQL data source."""
+
+    _inherit = "base.external.dbsource"
+
+    def connection_close_mysql(self, connection):
+        return connection.close()
+
+    def connection_open_mysql(self):
+        return sqlalchemy.create_engine(self.conn_string_full).connect()
+
+    def execute_mysql(self, sqlquery, sqlparams, metadata):
+        # FIXME: Duplicated method in modules to be consolidated in base
+        rows, cols = list(), list()
+        for record in self:
+            with record.connection_open() as connection:
+                if sqlparams is None:
+                    cur = connection.execute(sqlquery)
+                else:
+                    cur = connection.execute(sqlquery, sqlparams)
+                if metadata:
+                    cols = list(cur.keys())
+                rows = [r for r in cur]
```

Comment:
This is going to raise an exception if a query that doesn't return row is executed.
`rows = [r for r in cur] if cur.returns_rows else []` fixes this

Reply:
Thank you @mrm-david this could be proposed as a PR for `base_external_dbsource_sqlite` addon (which has the same issue and, as @yajo said, could be used instead of this MySQL addon). Could you make the PR?

Reply:
@dalonsod Would I need to write a unit test ? I have sadly very little experience with those.

Reply:
regarding to test coding, these addons are very particular, as mock objects are used to emulate tests, see 

https://github.com/solvosci/server-backend/blob/33de09e95bb8d8ef3340d50722b8777b69cfe630/base_external_dbsource_sqlite/tests/test_base_external_dbsource.py#L32-L48

Thus, your change won't imply tests update if I'm right.

Reply:
https://github.com/OCA/server-backend/pull/243